persistence-api/src/main/java/io/stargate/db/cdc/DefaultCDCHealthChecker.java

```diff
@@ -0,0 +1,136 @@
+  static final long TICK_INTERVAL = TimeUnit.SECONDS.toNanos(INTERVAL);
+
+  private AutoTickEWMA successes;
+  private AutoTickEWMA errors;
+
+  /**
+   * Creates a new instance of {@link DefaultCDCHealthChecker}.
+   *
+   * @param errorRateThreshold The percentage of error requests compared to total, expressed in as a
+   *     double from 0 to 1.
+   * @param minErrorsPerSecond The minimum amount of error occurrences per second for the health
+   *     checker to consider the error ratio. This setting is created to prevent detecting the CDC
+   *     producer as unhealthy when there's low traffic and few errors.
+   * @param ewmaIntervalMinutes The interval to determine the coefficient for the degree of
+   *     weighting decrease in the exponentially weighted moving average (EWMA). The health checker
+   *     will use this value to set a soothing factor equivalent to UNIX load average.
+   */
+  DefaultCDCHealthChecker(
+      double errorRateThreshold, int minErrorsPerSecond, int ewmaIntervalMinutes) {
+    this(errorRateThreshold, minErrorsPerSecond, ewmaIntervalMinutes, TickClock.defaultClock);
+  }
+
+  @VisibleForTesting
+  DefaultCDCHealthChecker(
+      double errorRateThreshold, int minErrorsPerSecond, int ewmaIntervalMinutes, TickClock clock) {
+
+    if (errorRateThreshold <= 0 || errorRateThreshold > 1) {
+      throw new IllegalArgumentException(
+          "Error rate threshold should be greater than 0 and lower than 1");
+    }
+
+    if (ewmaIntervalMinutes <= 0 || ewmaIntervalMinutes > 15) {
+      throw new IllegalArgumentException(
+          "The interval used to determine the smoothing factor for the exponentially "
+              + "weighted moving average must be higher 0 and lower than 15 minutes");
+    }
+
+    this.errorRateThreshold = errorRateThreshold;
+    this.minErrorsPerSecond = minErrorsPerSecond;
+    successes = new AutoTickEWMA(ewmaIntervalMinutes, clock);
+    errors = new AutoTickEWMA(ewmaIntervalMinutes, clock);
+  }
+
+  @Override
+  public boolean isHealthy() {
+    double errorRate = errors.getRate();
+    if (errorRate < minErrorsPerSecond || errorRate == 0.0) {
+      return true;
+    }
+
+    double successRate = successes.getRate();
+    System.out.println(String.format("Error: %f; Success: %f", errorRate, successRate));
+    double percentage = errorRate / (successRate + errorRate);
+    return percentage < errorRateThreshold;
+  }
+
+  @Override
+  public void reportSendError() {
+    errors.update();
+  }
+
+  @Override
+  public void reportSendSuccess() {
+    successes.update();
+  }
+
+  interface TickClock {
+    long getTick();
+
+    TickClock defaultClock =
+        new TickClock() {
+          private final Clock clock = com.codahale.metrics.Clock.defaultClock();
+
+          @Override
+          public long getTick() {
+            return clock.getTick();
+          }
+        };
+  }
+
+  @Override
+  public void close() {}
+
+  private static class AutoTickEWMA {
```

Comment:
I see that `EWMA` has a public constructor that allow to pass the tick interval:
```
public EWMA(double alpha, long interval, TimeUnit intervalUnit) {
        this.interval = intervalUnit.toNanos(interval);
        this.alpha = alpha;
    }
```
The default `private static final int INTERVAL = 5;` - so it is 5 seconds. But it seems that we can construct the EWMA passing 60 as the interval.

Reply:
That's the interval for each tick "step" of the decrease, the default of 5 seconds is ok.

Reply:
By this comment, I meant that maybe we don't need this custom implementation after all. It seems that the `EWMA` is solving our issue, and we can just change this parameter to `60` to achieve the same what `AutoTickEWMA` is doing. Or Am I missing something?

Reply:
EWMA will not tick by itself: https://github.com/dropwizard/metrics/blob/release/4.1.x/metrics-core/src/main/java/com/codahale/metrics/EWMA.java

It's up to the user of `EWMA` instances to call `tick` method: https://github.com/dropwizard/metrics/blob/release/4.1.x/metrics-core/src/main/java/com/codahale/metrics/Meter.java#L81

Reply:
What's the benefit of reporting the EWMA rather than the raw data and allowing the user to interpret the metrics as they want?